packages/nux/src/components/guide/test/finish-button.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import { create } from 'react-test-renderer';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import FinishButton from '../finish-button';
+
+describe( 'FinishButton', () => {
+	it( 'renders a button', () => {
+		const wrapper = shallow( <FinishButton /> );
+		expect( wrapper.find( Button ) ).toHaveLength( 1 );
+	} );
+
+	it( 'calls onClick when the button is clicked', () => {
+		const onClick = jest.fn();
+		const wrapper = shallow( <FinishButton onClick={ onClick } /> );
+		wrapper.find( Button ).prop( 'onClick' )();
```

Comment:
I think it tests the internals of React in the current form. It doesn't ensure that `onClick` is properly wired with the DOM.

Reply:
It's a unit test so we should only test the part that concerns `FinishButton`. In this case, we're testing that the `Button` component's `onClick` prop is wired to the `FinishButton` component's `onClick` prop. It's up to the `Button` component and its unit tests to ensure that the actual DOM event is handled.

(This test caught a bug while I was refactoring things in 78104f3db0190d6fac295933d44bee00a5d15c9f, too! 😀)